### DISCUSSION TITLE
Tilemaker ends without generating tiles with no error message

### DISCUSSION THREAD

**Discussion by satya-auti** [systemed/tilemaker#507] [General]
System configuration -> 512gb SSD (70gb available). 16gb RAM.
I have osm.pbf file with approx. 243mb, I have tried to generate the tiles. But the tilemaker ends without any error.
I used `--store` still I'm getting same.
@systemed is there any solution for this?

```
tilemaker --input bangalore-planet-v1.0.7.osm.pbf --bbox 76.474,12.0707,78.7149,13.8707 --output bangalore-planet-v1.0.7-tilemaker.mbtiles --process src/process.lua --config src/config.json --store /temp
Layer place (z0-14)
Layer boundary (z0-14)
Layer poi (z12-14)
Layer poi_detail (z14-14) -> poi
Layer housenumber (z14-14)
Layer waterway (z8-14)
Layer waterway_detail (z12-14) -> waterway
Layer transportation (z4-14)
Layer transportation_name (z8-14)
Layer transportation_name_mid (z12-14) -> transportation_name
Layer transportation_name_detail (z14-14) -> transportation_name
Layer building (z13-14)
Layer water (z6-14)
Layer ocean (z0-14) -> water
Layer water_name (z14-14)
Layer water_name_detail (z14-14) -> water_name
Layer aeroway (z11-14)
Layer aerodrome_label (z10-14)
Layer park (z11-14)
Layer landuse (z4-14)
Layer urban_areas (z4-8) -> landuse
Layer landcover (z0-14)
Layer ice_shelf (z0-9) -> landcover
Layer glacier (z2-9) -> landcover
Layer mountain_peak (z11-14)
Bounding box 76.474, 12.0707, 78.7149, 13.8707
Using osm store file: /temp
terminate called after throwing an instance of 'boost::filesystem::filesystem_error'
  what():  boost::filesystem::create_directory: Permission denied: "/temp"
[1]    3886 abort      tilemaker --input bangalore-planet-v1.0.7.osm.pbf --bbox  --output  --process
```

**Comment by systemed** [comment]
> boost::filesystem::create_directory: Permission denied: "/temp"

Is this not your problem - that you're trying to save into a new directory called `/temp` but running under a user that doesn't have permissions to create that directory?

Maybe try with `/tmp` instead, which probably exists already on your system?

**Reply by satya-auti** [reply]
@systemed I also used already exist folder then I got below error.
Sometimes tilemaker end without generating the tiles after using `--store` . (while running this I have only 10-11gb SSD available on my system.)
<img width="509" alt="image" src="https://github.com/systemed/tilemaker/assets/103890980/bc4bebef-e332-4318-8ede-06064074bbba">

**Reply by systemed** [reply]
Could you post the pbf somewhere so I can test using it?

**Reply by satya-auti** [reply]
https://www.dropbox.com/scl/fi/z57gbs9snrwnx4eimc6eo/mini-maharashtra-v1.0.7.osm.pbf?rlkey=a77l86fmigx2uorbukqxl4it9&dl=0

**Reply by systemed** [reply]
Thanks - what's your exact command line for that pbf?

**Reply by satya-auti** [reply]
`tilemaker --input mini-maharashtra-v1.0.7.osm.pbf --bbox 72.61,16.58,76.57,20.34 --output mini-maharashtra-v1.0.7.mbtiles --process src/process.lua --config src/config.json --store /tempstore`

**Reply by satya-auti** [reply]
@systemed are you able generate?

**Reply by systemed** [reply]
It looks like the .pbf was generated with osmium - what command line did you use to generate it?

**Reply by systemed** [reply]
Ah - you didn't say that it wasn't actually pure OSM data...

You have a relation 3000000000312 which contains way 3000000157124. tilemaker is intended for OSM data and will not process objects with IDs outside the normal range of OSM data - these very large IDs will bust its internal indexing.

You can either create the objects with natural IDs to begin with, or use `osmium renumber` before running tilemaker to get them back into a sane ID space.
